src/common/FastRational.h

```diff
@@ -385,6 +387,10 @@ class FastRational
         }
     }
 
+    bool isOneHeuristic() const {
```

Comment:
Maybe this could be just called `isOne()` ?

Reply:
I made this method because I needed a method that returns `true` if we can be sure it is `1`, but if it returns `false`, it does not necessarily means that it is not `1`.
To have a proper `isOne()` method, I would have to add also the check in case `wordPartValid()` returns `false`, similarly to `isZero()`.
On the other hand, I think we try to enforce the invariant that if the value fits into the wordPart that it always has `wordPartValid()`. I am just not sure if we do it in all places and that it is indeed an invariant in our current implementation.
So I went with this shortcut. But maybe the best solution would be just to go with proper `isOne` method. What do you think?

Reply:
Ok, yes.  It should be the case that whenever this external `isOneHeuristic()` method is called, the invariant "if the number fits in word representation, then `wordPartValid()` returns `true`" holds.  Maybe we should add an assert and check if this breaks.  It'd be useful to know if it doesn't hold.